server/src/telemetry/SentryServerTelemetry.ts

```diff
@@ -47,9 +47,17 @@ export class SentryServerTelemetry implements Telemetry {
       throw new Error("Tracing not loaded");
     }
 
+    const traceSampleRate = this.serverState.env === "development" ? 1 : 0.001;
+
     Sentry.init({
       dsn: this.dsn,
-      tracesSampleRate: serverState.env === "development" ? 1 : 0.001,
+      tracesSampler: ({ transactionContext: { name } }) => {
+        if (name === "indexing") {
```

Comment:
Nit suggestion:
I see we are using the raw `indexing` value in a few places already. I suggest exporting/reusing `INDEXING_JOB_ID` everywhere so that they don't go out of sync.